libcaf/libcaf/repository.py

```diff
@@ -803,12 +815,14 @@ def checkout(self, target: str) -> None:
         match resolved_ref:
             case SymRef():
                 write_ref(self.head_file(), resolved_ref)
+                return SymRef(f'{HEADS_DIR}/{target}')
```

Comment:
Why are you recreating the same ref you already have? Technically it's not a resolved_ref, which is what I suspect was confusing here